components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -146,6 +150,9 @@ open class GleanInternalAPI internal constructor () {
         Dispatchers.API.flushQueuedInitialTasks()
 
         // At this point, all metrics and events can be recorded.
+        // This should only be called from the main thread. This is enforced by
+        // the @MainThread decorator on this method.
```

Comment:
Does `@MainThread` break the build if called off the main thread? Or assert? Do we need to add `ThreadUtils.assertOnUiThread`? Moreover, can we add simple test coverage for this?

Reply:
It breaks the build.  Though, possibly only when called from Kotlin -- I'll need to confirm that.  I'm not sure how to write a test for a build failure.

It probably wouldn't hurt to add the assert as well.

Reply:
Ah, ok, if it breaks the build is fine. Don't mind the test.

Reply:
On further investigation, it seems to only break the build on some cases, in others, it's just a warning.  So I think doing both build time and run time checks seems safer.  And we can write a test for the run time check :)

Reply:
Great, thanks for checking that!